lib/workflows/dc-api-workflow.js

```diff
@@ -0,0 +1,331 @@
+  // /** @type {DcApi} */
+  // dcApi;
+
+  constructor(app) {
+    super(app);
+
+    app.get(
+      AUTHORIZATION_REQUEST_ENDPOINT,
+      this.authorizationRequest.bind(this),
+    );
+    app.post(
+      AUTHORIZATION_RESPONSE_ENDPOINT,
+      this.authorizationResponse.bind(this),
+    );
+
+    // Set the DC API Instance
+    this.initDcApi().then(dcApi => {
+      this.dcApi = dcApi;
+    });
+  }
+
+  async initDcApi() {
+    const rp = config.opencred.relyingParties.find(
+      rp => rp.workflow.type === 'dc-api',
+    );
+
+    const sk = config.opencred.signingKeys.find(k =>
+      k.purpose?.includes('authorization_request'),
+    );
+
+    const encoder = new TextEncoder();
+    const submissionEndpoint = rp.workflow.submissionEndpoint || '';
+    const referenceEndpoint = rp.workflow.referenceEndpoint || '';
+
+    const dcApi = await DcApi.new(
+      sk.privateKeyPem,
+      rp?.workflow?.baseUrl,
+      submissionEndpoint,
+      referenceEndpoint,
+      encoder.encode(config.opencred.caStore[0]),
+      this.createOID4VPSessionStore(),
+      this.createDCAPISessionStore(),
+    );
+
+    return dcApi;
+  }
+
+  async createWorkflowSpecificExchange(trustedVariables, untrustedVariables) {
+    if(trustedVariables.rp?.workflow?.type !== WORKFLOW_TYPE) {
+      return;
+    }
+
+    const dcApiSession = await this.dcApi.create_new_session();
+
+    const ex = await this.initExchange(trustedVariables, untrustedVariables);
+    if(!ex.workflowId) {
+      ex.workflowId = trustedVariables.rp.workflow.id;
+    }
+
+    ex.variables = {
+      dcApiSession,
+    };
+
+    await database.collections.Exchanges.insertOne(ex);
+
+    return this.formatExchange(ex);
+  }
+
+  async authorizationRequest(req, res) {
+    const rp = req.rp;
+    const exchange = await this.getExchange({rp, id: req.params.exchangeId});
+
+    logUtils.presentationStart(rp?.clientId, exchange?.id);
+    if(!exchange || exchange?.workflowId !== req.params.workflowId) {
+      const errorMessage = 'Exchange not found';
+      logUtils.presentationError(rp?.clientId, 'unknown', errorMessage);
+      res.status(404).send({message: errorMessage});
+      return;
+    }
+    if(exchange.state !== 'pending' && exchange.state !== 'active') {
+      const errorMessage = `Exchange in state ${exchange.state}`;
+      logUtils.presentationError(rp?.clientId, exchange.id, errorMessage);
+      res.status(400).send(errorMessage);
+      return;
+    }
+    try {
+      const sessionId =
+        exchange.variables.dcApiSession.session_creation_response.id;
+      const sessionSecret =
+        exchange.variables.dcApiSession.session_creation_response.client_secret;
+
+      if(!sessionSecret || !sessionId) {
+        res.status(500).send({message: 'Session data missing from exchange'});
+        return;
+      }
+
+      const requests = await this.dcApi.initiate_request(
+        sessionId,
+        sessionSecret,
+        JSON.parse(rp.workflow.dcApiRequest),
```

Comment:
No pre validation of the config, we'll see if we can come up with a schema to test against and provide nice error messages earlier in the config process.